src/ProgressiveCurve.sol

```diff
@@ -82,17 +82,18 @@ contract ProgressiveCurve is BaseCurve {
     /// @dev or to say that another way:
     /// $$\text{shares} = \sqrt{s^2 + \frac{2a}{m}} - s$$
     function previewDeposit(uint256 assets, uint256, /*totalAssets*/ uint256 totalShares)
-        public
+        external
         view
         override
         returns (uint256 shares)
     {
         require(assets > 0, "Asset amount must be greater than zero");
 
-        UD60x18 currentSupplyOfShares = UD60x18.wrap(totalShares);
```

Comment:
3.1 - Wrong Use of PRB-Math Library in Progressive Curves
Similar to the OffsetProgressiveCurve, we use `convert` instead of `wrap`, improving the accuracy of the curve as well as affecting the behavior of the slope.